File: core/shared/src/test/scala/diplomacy/variant/standard/VariantSpec.scala
package diplomacy.variant.standard

import diplomacy.UnitSpec
import diplomacy.variant.standard.map.Power
import diplomacy.variant.standard.map.Keywords._
import diplomacy.variant.standard.rule.StandardRuleUtils

class VariantSpec extends UnitSpec {
  "The variant" should {
    "define the initial board." in {
      val numberOfSupplyCenters = StandardRuleUtils.numberOfSupplyCenters(variant.initialBoard)
      for { power <- variant.initialBoard.map.powers } {
        val numOfSupplys = numberOfSupplyCenters(power)
        val numOfUnits = variant.initialBoard.units count { _.power == power }
        power match {
          case Russia =>
            numOfSupplys should be(4)
            numOfUnits should be(4)
          case _ =>
            numOfSupplys should be(3)
            numOfUnits should be(3)
        }
      }
    }
  }
}

File: test/variant/standard/variant-spec.ts
import * as chai from "chai"
import { variant } from "../../../src/variant/standard/module"
import { StandardRuleUtils } from "./../../../src/variant/standard/rule/module"
import { Power } from "./../../../src/variant/standard/map/module"

const should = chai.should()

describe("Variant", () => {
  it("defines an initial board.", () => {
    const numberOfSupplyCenters = StandardRuleUtils.numberOfSupplyCenters(variant.initialBoard)
    variant.initialBoard.map.powers.forEach(power => {
      const numOfUnits = ([...variant.initialBoard.units].filter(u => u.power === power)).length
      if (power === Power.Russia) {
        should.equal(numberOfSupplyCenters.get(power), 4)
        numOfUnits.should.equal(4)
      } else {
        should.equal(numberOfSupplyCenters.get(power), 3)
        numOfUnits.should.equal(3)
      }
    })
  })
})
